### PR TITLE
Added stuff to CoalGenerator

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description
added Tags
like i discussed you do unit tests <3
https://github.com/J3fftw1/LiteXpansion/pull/71#discussion_r492245657

## Changes
Added Tags

## Related Issues
N/A

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [x] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java [hunk 1]
```diff
@@ -44,14 +44,17 @@ public static Category getCategory(Plugin plugin, String name) {
...
         return new MockSlimefunItem(category, item, id);
     }
 
     public static VanillaItem mockVanillaItem(Plugin plugin, Material type, boolean enabled) {
-        Category category = new Category(new NamespacedKey(plugin, "test"), new CustomItem(Material.EMERALD, "&4Test Category"));
-        VanillaItem item = new VanillaItem(category, new ItemStack(type), type.name(), RecipeType.NULL, new ItemStack[9]);
+        Category category = new Category(new NamespacedKey(plugin, "test"),
+            new CustomItem(Material.EMERALD, "&4Test Category"));
+        VanillaItem item = new VanillaItem(category,
+            new ItemStack(type), type.name(), RecipeType.NULL, new ItemStack[9]);
```
Comment (TheBusyBiscuit) [inline src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:57]: Please let these two stay in one line.

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java [hunk 2]
```diff
@@ -44,14 +44,17 @@ public static Category getCategory(Plugin plugin, String name) {
     }
 
     public static SlimefunItem mockSlimefunItem(Plugin plugin, String id, ItemStack item) {
-        Category category = new Category(new NamespacedKey(plugin, "test"), new CustomItem(Material.EMERALD, "&4Test Category"));
+        Category category = new Category(new NamespacedKey(plugin, "test"),
+            new CustomItem(Material.EMERALD, "&4Test Category"));
```
Comment (TheBusyBiscuit) [inline src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:48]: This can stay as a one-liner

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java (hunk 1 above)
Comment (J3fftw1) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:57]: but formatting

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java (hunk 2 above)
Comment (J3fftw1) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:48]: but formatting

**Comment by LinoxGH** [comment]
Since this makes changea about tags, it should rather be committed to tags branch.

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:57]: Exactly...

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java (hunk 2 above)
Comment (TheBusyBiscuit) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:48]: Exactly...

**Comment by TheBusyBiscuit** [comment]
@LinoxGH Nope, this has nothing to do with the custom tags.

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java [hunk 3]
```diff
@@ -73,17 +73,64 @@ public static PlayerProfile awaitProfile(OfflinePlayer player) throws Interrupte
...
+        server.createMaterialTag(NamespacedKey.minecraft("wooden_buttons"), Material.BIRCH_BUTTON,
+            Material.ACACIA_BUTTON, Material.SPRUCE_BUTTON, Material.DARK_OAK_BUTTON,
+            Material.JUNGLE_BOAT, Material.OAK_BUTTON
+        );
+        server.createMaterialTag(NamespacedKey.minecraft("signs"), Material.SPRUCE_SIGN, Material.ACACIA_SIGN,
+            Material.OAK_SIGN, Material.JUNGLE_SIGN, Material.DARK_OAK_SIGN, Material.BIRCH_SIGN
+        );
+        server.createMaterialTag(NamespacedKey.minecraft("wooden_doors"), Material.SPRUCE_DOOR, Material.ACACIA_DOOR,
+            Material.OAK_DOOR, Material.JUNGLE_DOOR, Material.DARK_OAK_DOOR, Material.BIRCH_DOOR
+            );
```
Comment (TheBusyBiscuit) [inline src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:123]: Indentation is weird here

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java (hunk 3 above)
Comment (J3fftw1) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:123]: would you want 3 and 3 on a line

Comment (TheBusyBiscuit) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:123]: ?

Comment (TheBusyBiscuit) [reply src/test/java/io/github/thebusybiscuit/slimefun4/testing/TestUtilities.java:123]: I'm talking about the `);` that is at a different indentation level than the rest.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
WHY on earth did you set the base to `tags`...?

**Comment by TheBusyBiscuit** [comment]
Oh it was @LinoxGH... WHY would you do that...?

**Comment by TheBusyBiscuit** [comment]
@J3fftw1 I am closing this as it cannot be merged like that anymore..., please try to do it again without @LinoxGH fucking your PR up.

**Pull request closed without merging**